src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -394,17 +396,20 @@ private void HandleTopologyChange(DatabaseRecord newRecord)
                 DropOutgoingConnections(Destinations, instancesToDispose);
                 _internalDestinations.Clear();
                 _externalDestinations.Clear();
-                _destinations.Clear();
+                _destinations.Clear();                
                 DisposeConnections(instancesToDispose);
                 return;
             }
 
+            _clusterTopology = GetClusterTopology();
+
             HandleInternalReplication(newRecord, instancesToDispose);
             HandleExternalReplication(newRecord, instancesToDispose);
             var destinations = new List<ReplicationNode>();
             destinations.AddRange(_internalDestinations);
             destinations.AddRange(_externalDestinations);
-            _destinations = destinations;
+            _destinations = destinations;            
+            _numberOfSiblings = _destinations.Select(x => x.Url).Intersect(_clusterTopology.AllNodes.Select(x => x.Value)).Count();
```

Comment:
I don't think this is right. 
You are using the cluster topology, but a db isn't necessarily on all the nodes in the topology.